models/model_choice.py

```diff
@@ -27,41 +29,37 @@ def load_checkpoint(filename):
         raise FileNotFoundError(f"=> No model found at '{filename}'")
 
 
-def net(net_params, inference=False):
+def net(net_params, num_channels, inference=False):
     """Define the neural net"""
     model_name = net_params['global']['model_name'].lower()
-    num_classes = net_params['global']['num_classes']
-    if num_classes == 1:
-        warnings.warn("config specified that number of classes is 1, but model will be instantiated"
-                      " with a minimum of two regardless (will assume that 'background' exists)")
-        num_classes = 2
```

Comment:
manage num_classes parameter before getting into model_choice.py. Helps to make sure num_classes is uniform during execution.